src/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/FileStreamReadLightUp.cs

```diff
@@ -156,7 +156,7 @@ IntPtr overlapped
             );
 
             // older Windows systems:
-            [DllImport(@"kernel32.dll", EntryPoint = "ReadFile", ExactSpelling = true, SetLastError = true)]
+            [DllImport(@"api-ms-win-core-file-l1-1-0.dll", EntryPoint = "ReadFile", ExactSpelling = true, SetLastError = true)]
```

Comment:
> would really like to avoid this if at all possible. A lib like S.R.M shouldn't need to build separately for Windows and Unix when it is so far removed from the OS in its function.

The implementation currently has OS specific calls and so for now it can't be AnyCPU really. However depending on how #4786 is fixed it might change :). I will leave the right fix to you guys :)